packages/flutter/lib/src/cupertino/segmented_control.dart

```diff
@@ -98,11 +111,18 @@ class CupertinoSegmentedControl<T extends Object> extends StatefulWidget {
     this.selectedColor,
     this.borderColor,
     this.pressedColor,
+    this.disabledColor,
+    this.disabledTextColor,
     this.padding,
+    this.disabledChildren = const <Never>{},
   }) : assert(children.length >= 2),
        assert(
          groupValue == null || children.keys.any((T child) => child == groupValue),
          'The groupValue must be either null or one of the keys in the children map.',
+       ),
+       assert(
+         disabledChildren.length <= children.length,
```

Comment:
Is it possible that the disabled children length is larger than the children length? I think we don't have to assert this here. When the set contains one or more children, we disable the according segments; If the set contains other things, I think we don't need to care:)